src/PrestaShopBundle/DependencyInjection/CacheAdapterFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\DependencyInjection;
+
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Cache\Adapter\ApcuAdapter;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+
+/**
+ * Class CacheAdapterFactory responsible for returning the right Cache adapter for the associated driver
+ */
+class CacheAdapterFactory
+{
+    public function getCacheAdapter(string $driver): AdapterInterface
+    {
+        if ($driver === 'apcu') {
+            return new ApcuAdapter();
+        } elseif ($driver === 'memcached') {
+            return new MemcachedAdapter(
+                AbstractAdapter::createConnection('memcached://localhost', ['lazy' => true])
```

Comment:
I'm afraid we are not all using localhost for Memcached :thinking: 
Afaik, it's configurable in the BO

Reply:
I think you're right, it's configurable in the BO but I'm pretty sure that this configuration was only used in PrestaShop's own implementation of Memcached and not in the Doctrine's ones.

I can make it using it though!

Reply:
I'm not sure if it's the appropriate one but we have a service `prestashop.adapter.memcache_server.manager` that handles the memcache configs Maybe it's a good lead

Reply:
This would mean you may have to use DI actually to use this `CacheAdapterFactory` and it will have to be accessible in front container (unless you're gonna have to instanciate it manually anyway, since it is used in a CompilerPass so the container is not built yet 🤷‍♂️ )

Reply:
The problem with that is that the config is saved in the DB but we cannot access it using doctrine because we're actually configuring it at that moment 😅

Reply:
Should be fixed in another PR, the configuration has to be written when saving Cache configuration as we did it for the .htaccess.

Reply:
I created the issue: #25015